### ISSUE TITLE
[BUG] Pod stuck at pending phase after recurring backup with detached volume complete.

### ISSUE THREAD

**Issue by chriscchien** [longhorn/longhorn#3163]
**Describe the bug**
This issue **not happen every time**, but happened in automation test and can be manually reproduce.

When doing recurring backup on detached volume by scale deployment replica number to 0, while backup started, scale replica number to 1, sometimes after backup completed, pod will stuck at pending phase forever.

**To Reproduce**
1. Enable 'Allow Recurring Job While Volume Is Detached' in 'Setting' page
2. Setup S3 as back target in 'Setting' page
3. Create volume 'bak' with PV/PVC=bak, namespace=default
4. Create deployment by [deploy.zip](https://github.com/longhorn/longhorn/files/7365956/deploy.zip)
5. Volume should auto attached 
6. Write 400MB data to the volume from the pod `kubectl exec -it $podname -- /bin/bash -c "dd if=/dev/urandom of=/usr/share/nginx/html/data bs=1M count=400"` , wait write complete.
7. Scale deployment to 0, wait until volume detached.
8. Create a recurring job runs every 2 minutes. `*/2 * * * *`
9. Wait for backup to start.
10. While backup in progress, scale deployment replicas to 1.
11. Delete the recurring job
11. Pod stuck at 'Pending' phase after backing complete

**Expected behavior**
In step 12, after backup complete, pod should  be in running status

**Log**
[longhorn-support-bundle_59ebaf99-bb88-43c8-909f-8a36368bbaf2_2021-10-18T13-43-01Z.zip](https://github.com/longhorn/longhorn/files/7366097/longhorn-support-bundle_59ebaf99-bb88-43c8-909f-8a36368bbaf2_2021-10-18T13-43-01Z.zip)

Pod description
```
Name:           nginx-deployment-74d5586c54-5t86l
Namespace:      default
Priority:       0
Node:           worker1/192.168.122.138
Start Time:     Mon, 18 Oct 2021 21:14:29 +0800
Labels:         app=nginx
                pod-template-hash=74d5586c54
Annotations:    <none>
Status:         Pending
IP:             
IPs:            <none>
Controlled By:  ReplicaSet/nginx-deployment-74d5586c54
Containers:
  nginx:
    Container ID:   
    Image:          nginx:1.14.2
    Image ID:       
    Port:           80/TCP
    Host Port:      0/TCP
    State:          Waiting
      Reason:       ContainerCreating
    Ready:          False
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /usr/share/nginx/html from mypod (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-drx8d (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  mypod:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  bak
    ReadOnly:   false
  kube-api-access-drx8d:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason              Age                    From                     Message
  ----     ------              ----                   ----                     -------
  Warning  FailedMount         15m (x9 over 105m)     kubelet                  Unable to attach or mount volumes: unmounted volumes=[mypod], unattached volumes=[kube-api-access-drx8d mypod]: timed out waiting for the condition
  Warning  FailedAttachVolume  5m57s (x57 over 106m)  attachdetach-controller  AttachVolume.Attach failed for volume "bak" : rpc error: code = DeadlineExceeded desc = volume bak failed to attach to node worker1
  Warning  FailedMount         3m57s (x37 over 103m)  kubelet                  Unable to attach or mount volumes: unmounted volumes=[mypod], unattached volumes=[mypod kube-api-access-drx8d]: timed out waiting for the condition

```

**Environment:**
 - Longhorn version: v1.2.x-head
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): Kubectl
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: v1.21.5+k3s2
   - Number of management node in the cluster: 1
   - Number of worker node in the cluster: 3
 - Node config
   - OS type and version: Ubuntu 20.04.3
   - CPU per node: 4
   - Memory per node: 4G
   - Disk type(e.g. SSD/NVMe): SSD
   - Network bandwidth between the nodes: Local VM network
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): multipass + libvirt
 - Number of Longhorn volumes in the cluster: 1

**Additional context**
This error not happen easily, but when this error happened, from UI I can see volume attached to node, state was healthy and all replica running.
But from node terminal I can not see anything in **/dev/longhorn/** (no matter automation test or manually reproduce scenario)

Automation test case :test_recurring_job.py::test_recurring_jobs_allow_detached_volume
[Jenkins related fail log](https://ci.longhorn.io/job/public/job/v1.2.x/job/v1.2.x-lh-tests-ubuntu-amd64/22/testReport/junit/tests/test_recurring_job/test_recurring_jobs_allow_detached_volume_s3_/)

**Comment by wonleing** [comment]
I can reproduce this kind of error (Unable to attach or mount volumes) in many different ways:
1. start 20+ vms at the same time
2. clone 20+ dvs at the same time
3. clone a dv with a dv bounded in started VM.

These errors don't exist when I use PVC and local storage. So it  must be related to longhorn.

**Comment by wonleing** [comment]
Error logs:

kubelet[24628]: E1228 14:46:29.159943   24628 nestedpendingoperations.go:301] Operation for "{volumeName:kubernetes.io/csi/driver.longhorn.io^pvc-c9af6cef-f52e-49a4-81ef-41fa3dc4ad17 podName: nodeName:}" failed. No retries permitted until 2021-12-28 14:48:31.159923141 +0800 CST m=+7687.524610133 (durationBeforeRetry 2m2s). Error: "UnmapDevice failed for volume \"pvc-c9af6cef-f52e-49a4-81ef-41fa3dc4ad17\" (UniqueName: \"kubernetes.io/csi/driver.longhorn.io^pvc-c9af6cef-f52e-49a4-81ef-41fa3dc4ad17\") on node \"k8s-master\" : The device \"/var/lib/kubelet/plugins/kubernetes.io/csi/volumeDevices/pvc-c9af6cef-f52e-49a4-81ef-41fa3dc4ad17/dev\" is still referenced from other Pods [/var/lib/kubelet/plugins/kubernetes.io/csi/volumeDevices/pvc-c9af6cef-f52e-49a4-81ef-41fa3dc4ad17/dev/51054e19-3773-45ba-8ad3-be2c7e54a458]"

kubelet[24628]: E1228 14:46:31.675721   24628 nestedpendingoperations.go:301] Operation for "{volumeName:kubernetes.io/csi/driver.longhorn.io^pvc-ed584b27-ac97-4100-b6ff-ac2e40d6c99d podName: nodeName:}" failed. No retries permitted until 2021-12-28 14:48:33.67570173 +0800 CST m=+7690.040388724 (durationBeforeRetry 2m2s). Error: "UnmapDevice failed for volume \"pvc-ed584b27-ac97-4100-b6ff-ac2e40d6c99d\" (UniqueName: \"kubernetes.io/csi/driver.longhorn.io^pvc-ed584b27-ac97-4100-b6ff-ac2e40d6c99d\") on node \"k8s-master\" : The device \"/var/lib/kubelet/plugins/kubernetes.io/csi/volumeDevices/pvc-ed584b27-ac97-4100-b6ff-ac2e40d6c99d/dev\" is still referenced from other Pods [/var/lib/kubelet/plugins/kubernetes.io/csi/volumeDevices/pvc-ed584b27-ac97-4100-b6ff-ac2e40d6c99d/dev/2faa0298-f333-4a8f-a387-cc1c955cbf4f]"

**Comment by PhanLe1010** [comment]
@wonleing 

The issues you mentioned doesn't seem related to this issue since this one is about recurring backup job.


Can you create a new issue with more detail about the reproducing steps? You are saying VM and DV, are you referring to Harvester?

**Comment by wonleing** [comment]
> @wonleing
> 
> The issues you mentioned doesn't seem related to this issue since this one is about recurring backup job.
> 
> Can you create a new issue with more detail about the reproducing steps? You are saying VM and DV, are you referring to Harvester?

right. The bug is here: https://github.com/harvester/harvester/issues/1754
